pkgs/applications/graphics/qvge/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, mkDerivation
+, fetchFromGitHub
+, substituteAll
+, qmake
+, qtx11extras
+, graphviz
+}:
+
+mkDerivation rec {
+  pname = "qvge";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "ArsMasiuk";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0j4ih03nl6iihhnxrfldkarg9hvxb62lpr58xspn417d3gj6xjll";
+  };
+
+  sourceRoot = "${src.name}/src";
```

Comment:
```suggestion
  prePatch = ''
    cd src
  '';
```

Reply:
Why "cd src" is better here?

Reply:
There are a few reasons:

1. It won't be required to update the `sourceRoot` when you override `src`
2. Relying on `src.name` will not work if we switch to fetching tarball, it won't work for filesystem paths either

Reply:
I see, fixed.